src/components/WearablePreview/WearablePreview.tsx

```diff
@@ -0,0 +1,371 @@
+        skinParam,
+        hairParam,
+        eyesParam,
+        bodyShapeParam,
+        emoteParam,
+        cameraParam,
+        projectionParam,
+        zoomParam,
+        backgroundParam,
+        offsetXParam,
+        offsetYParam,
+        offsetZParam,
+        cameraXParam,
+        cameraYParam,
+        cameraZParam,
+        wheelZoomParam,
+        wheelPrecisionParam,
+        wheelStartParam,
+        disableBackgroundParam,
+        disableAutoRotateParam,
+        disableAutoCenterParam,
+        disableFaceParam,
+        disableDefaultWearablesParam,
+        disableDefaultEmotesParam,
+        disableFadeEffectParam,
+        showSceneBoundariesParam,
+        showThumbnailBoundariesParam,
+        peerUrlParam,
+        nftServerUrlParam,
+        typeParam,
+        panningParam,
+        lockAlphaParam,
+        lockBetaParam,
+        lockRadiusParam,
+        envParam,
+        unityParam,
+        unityModeParam,
+      ]
+        .filter((param) => !!param)
+        .join("&")
+    return url
+  }, [restProps, baseUrl, dev, unity])
+
+  const getOptions = useCallback(() => {
+    const options: PreviewOptions = {}
+
+    for (const key in restProps) {
+      if (typeof restProps[key] !== "function") {
+        options[key] = restProps[key]
+      }
+    }
+
+    return options
+  }, [restProps])
+
+  const sendUpdate = useCallback(
+    (options: PreviewOptions) => {
+      const { lastOptions } = state
+      // only send update if new options are different
+      if (!lastOptions || !equal(options, lastOptions)) {
+        // send message to iframe
+        if (iframeRef.current?.contentWindow) {
+          sendMessage(
+            iframeRef.current.contentWindow,
+            PreviewMessageType.UPDATE,
+            {
+              options,
+            }
+          )
+        }
+        // callback
+        onUpdate(options)
+
+        // store options on state
+        setState((prev) => ({ ...prev, lastOptions: options }))
+      }
+    },
+    [state, onUpdate]
+  )
+
+  const handleUpdate = useCallback(() => {
+    if (iframeRef.current) {
+      // SSR check
+      if (typeof window !== "undefined") {
+        const options = getOptions()
+        if (state.isReady) {
+          // if the iframe is ready, send the update
+          sendUpdate(options)
+        } else {
+          // otherwise store last update in state until it's ready
+          setState((prev) => ({ ...prev, pendingOptions: options }))
+        }
+      }
+    } else {
+      console.warn(`Could not send update, iframe is not referenced`)
+    }
+  }, [getOptions, state.isReady, sendUpdate])
+
+  const handleMessage = useCallback(
+    (event: MessageEvent) => {
+      const { origin } = event
+      if (origin === baseUrl) {
+        if (event.data && event.data.type) {
+          const type: PreviewMessageType = event.data.type
+          switch (type) {
+            case PreviewMessageType.LOAD: {
+              onLoad()
```

Comment:
can we send here the renderer loaded from the event payload?
https://github.com/decentraland/ui/blob/master/src/components/WearablePreview/WearablePreview.tsx#L327-L332